internal/terraform/context_plan.go

```diff
@@ -419,11 +419,10 @@ func (c *Context) planWalk(config *configs.Config, prevRunState *states.State, r
 	// we can now walk.
 	changes := plans.NewChanges()
 	walker, walkDiags := c.walk(graph, walkOp, &graphWalkOpts{
-		Config:             config,
-		InputState:         prevRunState,
-		Changes:            changes,
-		MoveResults:        moveResults,
-		RootVariableValues: rootVariables,
```

Comment:
I had indeed intended to remove that entirely here, so that the variable values are used directly only during graph _construction_ and not during the walk. However, I can see now that I got a bit lost in the noise and neglected to clean up the top-level mentions of it, even though (as you noted) the deeper stuff isn't referring to it anymore. Thanks! I'll see about cleaning that up.

Reply:
Oh, looking closer I notice that the change you suggested is valid but not for the reason I explained in my previous comment: `planWalk` _does_ include the graph construction and so it does still need the variables, but it now gets them from the `opts *PlanOpts` argument along with all of the other plan options, rather than needing a separate argument to carry them.